app/main.py

```diff
@@ -2,24 +2,78 @@
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, instance: None, name: str) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: None, objtype: None) -> None:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: None, value: int) -> None:
```

Comment:
typehints are invalid
https://stackoverflow.com/questions/3798835/understanding-get-and-set-and-python-descriptors